libsolidity/interface/StandardCompiler.cpp

```diff
@@ -1764,34 +1728,6 @@ Json StandardCompiler::compile(Json const& _input) noexcept
 		else
 			return formatFatalError(Error::Type::JSONError, "Only \"Solidity\", \"Yul\", \"SolidityAST\" or \"EVMAssembly\" is supported as a language.");
 	}
-	catch (Json::parse_error const& _exception)
-	{
-		return formatFatalError(Error::Type::InternalCompilerError, std::string("JSON parse_error exception: ") + util::removeNlohmannInternalErrorIdentifier(_exception.what()));
-	}
-	catch (Json::invalid_iterator const& _exception)
-	{
-		return formatFatalError(Error::Type::InternalCompilerError, std::string("JSON invalid_iterator exception: ") + util::removeNlohmannInternalErrorIdentifier(_exception.what()));
-	}
-	catch (Json::type_error const& _exception)
-	{
-		return formatFatalError(Error::Type::InternalCompilerError, std::string("JSON type_error exception: ") + util::removeNlohmannInternalErrorIdentifier(_exception.what()));
-	}
-	catch (Json::out_of_range const& _exception)
-	{
-		return formatFatalError(Error::Type::InternalCompilerError, std::string("JSON out_of_range exception: ") + util::removeNlohmannInternalErrorIdentifier(_exception.what()));
-	}
-	catch (Json::other_error const& _exception)
-	{
-		return formatFatalError(Error::Type::InternalCompilerError, std::string("JSON other_error exception: ") + util::removeNlohmannInternalErrorIdentifier(_exception.what()));
-	}
-	catch (Json::exception const& _exception)
-	{
-		return formatFatalError(Error::Type::InternalCompilerError, std::string("JSON runtime exception: ") + util::removeNlohmannInternalErrorIdentifier(_exception.what()));
-	}
```

Comment:
Note also that by removing those we're not losing any details. In the generic exception handler we'll still get the full message and also diagnostics saying where it was thrown from. For an ICE the message does not need to be sanitized with `removeNlohmannInternalErrorIdentifier()`, because it can be considered diagnostic info too. In fact, the more internal details the better.

This sanitization is still good for proper `JSONError`s though, because those can appear in test expectations and are considered a part of our interface.